includes/TypesForLines/LineMeshRefinement.js

```diff
@@ -1,8 +1,163 @@
+/**
+* Creates line mesh refinement
+* @functiom
+* @constructor
+* @param	{Number}			no						Index of line mesh refinement, can be undefined
+* @param	{Number}			target_length			Length of FEA element side on the line, can be undefined
+* @param	{Number}			layers					Number of layers to be affected by this refinement, can be undefined
+* @param	{Number or Array}	lines					Lines assigned to this refinement, can be undefined
+* @param	{String}			comment					Comment, can be undefined
+* @param	{Object}			params					line mesh refinement parameters, can be undefined
+* @return	{Object}			Created line mesh refinement
+*/
+var createLineMeshRefinement = function (no,
+										 target_length,											
+										 layers,
+										 lines,
+										 comment,
+										 params) {
+
+
+
+	if (no != undefined) {
+		var lineMeshRefinement = line_mesh_refinements.create(no);
+	}
+	else {
+	 	var lineMeshRefinement = line_mesh_refinements.create();
+	}
+
+	if (target_length != undefined) {
+		lineMeshRefinement.target_length = target_length;
+	}
+
+	if (layers != undefined) {
+		lineMeshRefinement.number_of_layers = layers;
+	}
+
+	if (lines != undefined) {
+		lineMeshRefinement.lines = lines;
+	}
+	set_comment_and_parameters(lineMeshRefinement, comment, params);
+
+
+	return lineMeshRefinement;
+};
+
+/**
+* Creates line mesh refinement
+* @class
+* @constructor
+* @param	{Number}			no						Index of line mesh refinement, can be undefined
+* @param	{Number}			target_length			Length of FEA element side on the line, can be undefined
+* @param	{Number}			layers					Number of layers to be affected by this refinement, can be undefined
+* @param	{Number or Array}	lines					Lines assigned to this refinement, can be undefined
+* @param	{String}			comment					Comment, can be undefined
+* @param	{Object}			params					line mesh refinement parameters, can be undefined
+* @return	{Object}			Created line mesh refinement HLF
+*/
 function LineMeshRefinement(no,
+							target_length,
+							layers,
+							lines,
                             comment,
                             params)
 {
-    var line_mesh_refinement = engine.create_line_mesh_refinement(no);
-    set_comment_and_parameters(line_mesh_refinement, comment, params);
-    return line_mesh_refinement;
+    this.settings = createLineMeshRefinement(no, target_length, layers, lines, comment, params);
```

Comment:
add condition if(arguments.lenght !==0){} and put code inside this block. User could have option to create just LineMeshRefinement object and then use one function from the three main types.

Reply:
it is already possible